base/tuple.jl

```diff
@@ -135,6 +135,33 @@ rest(t::Tuple, i::Int) = ntuple(x -> getfield(t, x+i-1), length(t)-i+1)
 rest(a::Array, i::Int=1) = a[i:end]
 rest(itr, state...) = Iterators.rest(itr, state...)
 
+_collect_n(itr, ::Val{0}, st...) = ((), st)
+function _collect_n(itr, ::Val{N}, st...) where {N}
+    tmp = iterate(itr, st...)
+    if tmp === nothing
+        error("Iterator does not contain enough elements for the given variables.")
+    end
+    first, st′ = tmp
+    tail, st′′ = _collect_n(itr, Val(N-1), st′)
+    return (first, tail...), st′′
+end
+
+function split_rest(itr, ::Val{N}, st...) where {N}
+    if IteratorSize(itr) == IsInfinite()
```

Comment:
What about `SizeUnknown()`? That may be finite or infinite.

Reply:
We can't really detect that in this case, so `split_rest` will still try to collect it. If it is actually infinite this will mean that `split_rest` gets stuck in an infinite loop, but I believe that's the case for lots of other functions on iterators already ,so I wouldn't worry about that too much.